bot/common/threads/onboarding.py

```diff
@@ -98,56 +118,315 @@ async def handle_emoji(self, raw_reaction):
+            "Could not find tweet with supplied id %s in %s's twitter history."
+            " Please make sure that you tweeted in the last %s minutes, and the"
+            " verification tweet is among your %s most recent."
+        ) % (status_id, profile, MAX_TWEET_LOOKBACK_MINUTES, MAX_TWEETS_TO_RETRIEVE)
+        raise ThreadTerminatingException(err_msg)
+
+    return kmalone_tweet
+
+
+def verify_tweet_text(tweet_text, expected_tweet_text):
+    if tweet_text != expected_tweet_text:
+        raise ThreadTerminatingException(
+            "Tweet text %s doesn't match the verification tweet %s"
+            % (tweet_text, expected_tweet_text)
+        )
+
+
+class AddUserWalletAddressStep(AddUserAccountStep):
+    """Step to submit wallet address for the govrn profile"""
+
+    name = StepKeys.ADD_USER_WALLET_ADDRESS.value
+
+    def __init__(self, user_id, guild_id, cache):
+        super().__init__(user_id, guild_id, cache)
+
+    def get_account_descriptor(self):
+        return "Ethereum wallet address"
+
+    def sanitize_account_input(self, message):
+        stripped_message = message.content.strip()
+        if not Web3.isAddress(stripped_message):
+            raise ThreadTerminatingException(
+                "Supplied address %s is not a valid ethereum address" % stripped_message
+            )
+
+        return stripped_message
+
+    def get_account_storage_key(self):
+        return WALLET_STORAGE_KEY
 
     async def handle_emoji(self, raw_reaction):
         return _handle_skip_emoji(raw_reaction, self.guild_id)
 
 
+class PromptUserWalletMessageSignatureStep(PromptForAccountVerificationStep):
+    """Prompts the user to sign a message with their specified wallet"""
+
+    name = StepKeys.PROMPT_USER_WALLET_ADDRESS.value
+
+    def __init__(self, user_id, guild_id, cache):
+        super().__init__(user_id, guild_id, cache)
+
+    def get_account_verification_prompts(self):
+        instructions_embed = discord.Embed(
+            colour=INFO_EMBED_COLOR,
+            title="Verification instructions",
+            description=WALLET_VERIFICATION_INSTRUCTIONS,
+        )
+
+        prompt = discord.Embed(
+            colour=INFO_EMBED_COLOR,
+            title="Please sign the below message with your wallet",
+            description=REQUESTED_SIGNED_MESSAGE,
+        )
+        return [instructions_embed, prompt]
+
+
+class VerifyUserWalletMessageSignatureStep(VerifyAccountStep):
+    """Verifies the message supplied by the user was signed by their specified wallet"""
+
+    name = StepKeys.VERIFY_USER_WALLET_ADDRESS.value
+
+    def __init__(self, user_id, guild_id, cache):
+        super().__init__(user_id, guild_id, cache)
+
+    async def verify_account(self, authentication_message):
+        address = await get_cache_metadata(self.user_id, self.cache, WALLET_STORAGE_KEY)
+        stripped_supplied_signature = authentication_message.content.strip()
+
+        try:
+            int(stripped_supplied_signature)
```

Comment:
Is this either a none or an int? It may be better to handle this control flow with an if statement